mlrun/frameworks/lgbm/callbacks/logging_callback.py

```diff
@@ -0,0 +1,109 @@
+        :param static_hyperparameters:  If needed to track a hyperparameter one time per run it should be passed here.
+                                        The parameter expects a list of all the hyperparameters names to track our of
+                                        the `params` dictionary.
+        """
+        super(LoggingCallback, self).__init__()
+        self._logger = Logger()
+        self._dynamic_hyperparameters_keys = (
+            dynamic_hyperparameters if dynamic_hyperparameters is not None else {}
+        )
+        self._static_hyperparameters_keys = (
+            static_hyperparameters if static_hyperparameters is not None else {}
+        )
+
+    @property
+    def logger(self) -> Logger:
+        """
+        Get the logger of the callback. In the logger you may access the collected training data.
+
+        :return: The logger.
+        """
+        return self._logger
+
+    def __call__(self, env: CallbackEnv):
+        """
+        Log the iteration that ended and all the results it calculated.
+
+        :param env: A named tuple passed ad the end of each iteration containing the metrics results. For more
+                    information check the `Callback` doc string.
+        """
+        # Log the iteration:
+        self._logger.log_iteration()
+
+        # Log the metrics results out of the `evaluation_result_list` field:
+        self._log_results(evaluation_result_list=env.evaluation_result_list)
+
+        # Log the hyperparameters out of the `params` field:
+        self._log_hyperparameters(parameters=env.params)
+
+    def _log_results(
+        self, evaluation_result_list: List[LGBMTypes.EvaluationResultType]
+    ):
+        """
+        Log the callback environment results data into the logger.
+
+        :param evaluation_result_list: The metrics results as provided by the callback environment of LightGBM.
+        """
+        for evaluation_result in evaluation_result_list:
+            # Check what results were given, from `lightgbm.train` or `lightgbm.cv`:
+            if len(evaluation_result) == 4:
+                # `lightgbm.train` is used:
+                self._logger.log_result(
+                    validation_set_name=evaluation_result[0],
+                    metric_name=evaluation_result[1],
+                    result=evaluation_result[2],
+                )
+            else:
+                # `lightgbm.cv` is used, unpack both mean and stdv scores:
+                self._logger.log_result(
+                    validation_set_name=evaluation_result[0],
+                    metric_name=f"{evaluation_result[1]}_mean",
+                    result=evaluation_result[2],
+                )
+                self._logger.log_result(
+                    validation_set_name=evaluation_result[0],
+                    metric_name=f"{evaluation_result[1]}_stdv",
+                    result=evaluation_result[4],
```

Comment:
Lets try to avoid magic numbers, I am not quite sure why you are checking exactly for the length of 4.
What is expected to happen length is greater than 4?
Most of the times when we use numbers and it is not obvious why, we assign the number to a parameter which clarifies what is that index referred to.